src/reactor/registration.rs

```diff
@@ -0,0 +1,185 @@
+#[cfg(windows)]
```

Comment:
imo it would be a good idea to split this into one file per differing implementation in order to make it easier to compare them side-by-side. (and maybe replace the module hack by imports via paths)